### PR TITLE
[WIP] Code refactoring

### PR DESCRIPTION
**Pull request by KittyGiraudel**
To be added:
- [x] Preserve extension on output
- [x] Update (rewrite?) tests
- [x] Update README

(From #16, because I failed at Git. @valeriangalliat, you now have full rights over the repository.)

### CODE REVIEW

File: lib/brick.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
...
...ICAgICAKICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIAogICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgCiAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAKICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIAogICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgCiAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAKICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIAogICAgICAgICAgICAgICAgICAgICAgICAgICAgCjw/eHBhY2tldCBlbmQ9InciPz5U92F4AAAAIGNIUk0AAHolAACAgwAA+f8AAIDpAAB1MAAA6mAAADqYAAAXb5JfxUYAAAX6SURBVHjadJdbT9tKFIU/j2d8iROHmIu4lNIi3pCq/rP+yUrtY1WBoKIFtaVARIDQJL6NL+fhnJmTpGkkS4nj2Wtf1tp723n37l3b7XapqorlT9u2f/3d
```
Comment (pascalduez) [inline lib/brick.js]: Not sure this is a valid ES2015 modules export.

Comment (KittyGiraudel) [inline lib/brick.js]: It is. :)

**Comment by valeriangalliat** [comment]
- Updated according to previous PR changes.
- Rewrote tests to not assert only proper behavior and output, without caring about internal implementation.
- Removed `images/sample.lego.jpg`.
- Updated `lib/brick.js` to get the image from `images/brick.png` (previously unused) instead of hardcoded Base64 string.
- Added ability to define the output format.
- Tweaks in `package.json` including `test` command simplification, and dependencies maintenance.

**Comment by valeriangalliat** [comment]
For the readme, document how to process file globs using `vinyl-fs` and https://www.npmjs.com/package/vinyl-transform.

File: lib/brick.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import fs from 'fs'
+import path from 'path'
+import Canvas from 'canvas'
+
+export default Object.assign(new Canvas.Image(), {
+  src: fs.readFileSync(path.resolve(__dirname, '../images/brick.png'))
```
Comment (KittyGiraudel) [inline lib/brick.js:6]: Why do you actually read the file here? All you need is the path, no?

File: package.json [hunk 3]
```diff
@@ -14,26 +14,29 @@
...
+    "multipipe": "^0.2.1",
+    "readable-stream": "^2.0.4"
   },
   "scripts": {
-    "start": "npm run dist && npm run test",
-    "dist": "babel ./lib --out-dir ./dist",
-    "lint": "semistandard",
-    "demo": "rm -f ./images/sample.lego.png && ./bin/node-legofy ./images/sample.jpg",
-    "test": "mocha ./test/**/*.js --compilers js:babel-core/register"
+    "demo": "bin/node-legofy < images/sample.jpg > images/sample.lego.jpg",
```
Comment (KittyGiraudel) [inline package.json:25]: This should be called `sample` I guess.

File: package.json [hunk 4]
```diff
@@ -16,16 +16,19 @@
...
-    "demo": "rm -f ./images/sample.lego.png && ./bin/node-legofy ./images/sample.jpg",
-    "test": "mocha ./test/**/*.js --compilers js:babel-core/register"
+    "dist": "babel lib --out-dir dist",
+    "lint": "standard lib",
+    "test": "bin/node-legofy < images/sample.jpg > images/sample.lego.jpg"
   },
-  "semistandard": {
+  "standard": {
     "parser": "babel-eslint",
     "ignore": [
```
Comment (KittyGiraudel) [inline package.json:33]: This should be removed as we don’t use `mocha` anymore.

File: package.json [hunk 5]
```diff
@@ -14,26 +14,29 @@
...
+    "prepublish": "npm run lint && npm run dist && npm run test",
+    "test": "mocha test --compilers js:babel-core/register"
   },
-  "semistandard": {
+  "standard": {
     "parser": "babel-eslint",
     "ignore": [
       "dist"
     ],
     "globals": [
```
Comment (KittyGiraudel) [inline package.json:36]: This is recommended to be installed globally by Babel.

File: lib/brick.js (hunk 2 above)
Comment (valeriangalliat) [inline lib/brick.js:6]: You're right.

File: package.json (hunk 3 above)
Comment (valeriangalliat) [inline package.json:25]: Sure.

File: package.json (hunk 4 above)
Comment (valeriangalliat) [inline package.json:33]: We do still use `mocha`

**Pull request merged**